yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+
+Finally, all of these project components, including architecture documentation,
+features documentation, build environment, tests, and CI/CD, collectively
+contribute to ongoing maintenance and updates. They form a cohesive framework
+that supports the project throughout its lifecycle, enabling us to address
+security vulnerabilities, minimize technical debt, and integrate new
+functionalities as we maintain and update our Yocto-based projects.
+
+---
+# Update strategy
+
+* Remembering every project component during updates
+  - each project component, such as architecture documentation, features
+    documentation, build environment, tests, and CI/CD, plays a crucial role in
+    the project's functionality and stability
+  - updates must consider all components to ensure compatibility, minimize
+    regressions, and maintain the overall integrity of the Yocto-based project
+* Importance of the update deployment method
+  - the update deployment method impacts the efficiency, scalability, and user
+    experience of the Yocto-based project
+  - having a dedicated update system allows for streamlined and controlled
+    updates, minimizing downtime and reducing the need for manual intervention
+  - evaluating the requirements and constraints of the project helps determine
+    whether a dedicated update system is necessary or if reflashing devices is
+    feasible
+
+???
+
+TBD notes
+
+---
+# Update strategy
+
+* Yocto release cycle
+ - updating based on the Yocto release cycle ensures access to the latest
+   features, bug fixes, and security patches provided by the Yocto Project.
+ - regularly synchronizing with the Yocto release cycle maintains a stable and
+   up-to-date project, leveraging the community's collective efforts and
+   expertise
+* CVE checks and security reports
+  - integrating CVE (Common Vulnerabilities and Exposures) checks into the Yocto
+    build helps identify security vulnerabilities in the project's components
+  - organizing updates based on security reports allows for timely patching and
+    protection against known vulnerabilities
+* Feature-set requirements
+ - considering feature updates allows for tailored enhancements, ensuring the
+   project meets evolving functional needs and remains competitive in the market
+
+???
+
+TBD notes
+https://docs.yoctoproject.org/ref-manual/release-process.html
+
+---
+# Performing update
+
+* Build environment
+  - ensuring the build environment is up-to-date and compatible with the Yocto
+    Project requirements
+* External layers update
+  - updating external layers used in the project to their latest versions
+* Custom layer update
+  - updating the custom layer(s) specific to the project, incorporating
+    necessary changes and improvements
+* Real-life examples
+  - responsive multimedia casting device using RaspberryPi
+  - Yocto BSP maintenance
+  - secure gateway for IoT data transfer using i.MX6
+  - full build automation in project using Rockchip RK3399
```

Comment:
This slides looks much better in general, than two previous ones, in terms of text capacity

Reply:
Roger that